src/Controls/tests/Xaml.UnitTests/AssemblyInfoTests.cs

```diff
@@ -71,28 +74,41 @@ public void ProductAndCompany(string assemblyName)
 			Assert.AreEqual(s_company, actual.CompanyName);
 		}
 
-		static string GetFileFromRoot(string file)
+		internal static string GetFilePathFromRoot(string file)
```

Comment:
@mattleibow @jonathanpeppers  you have any ideas to improve these tests where we need to reference the root directory , seems this is really easy to break.

Reply:
It's not very pretty, but you could make an MSBuild target write `$(MSBuildThisFileDirectory)` in a `.cs` file.

Something like:
```xml
<!-- untested, from memory -->
<WriteLinesToFile File="$(IntermediateOutputPath)TestConstants.cs" Lines="class TestConstants { public const string Directory = &quot;$(MSBuildThisFileDirectory)&quot;; }" WriteOnlyWhenDifferent="true" />
<ItemGroup>
  <Compile Include="$(IntermediateOutputPath)TestConstants.cs" />
  <FileWrites Include="$(IntermediateOutputPath)TestConstants.cs" />
</ItemGroup>
```

If running from the `.csproj`, this would point to its directory.